include/rmm/mr/device/limiting_resource_adaptor.hpp

```diff
@@ -130,11 +132,12 @@ class limiting_resource_adaptor final : public device_memory_resource {
     void* p = nullptr;
 
     std::size_t proposed_size = rmm::detail::align_up(bytes, allocation_alignment_);
-    if (proposed_size + allocated_bytes_ <= allocation_limit_) {
+    allocated_bytes_ += proposed_size;
+    if (allocated_bytes_ <= allocation_limit_) {
```

Comment:
Apart from the fact that this is doing a redundant atomic load, I don't believe this is the semantics of what we want. This allows an intervening thread to impact the value of `allocated_bytes_` between the `fetch_add` and the `load`.

Reply:
Fixed.